gmso/lib/potential_templates.py

```diff
@@ -1,158 +1,130 @@
 
         super(PotentialTemplate, self).__init__(
             name=name,
             expression=expression,
             independent_variables=independent_variables,
-            template=True,
-        )
-
-class HarmonicTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='HarmonicTorsionPotential',
-                 expression='0.5 * k * (phi - phi_eq)**2',
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class PeriodicTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='PeriodicTorsionPotential',
-                 expression='k * (1 + cos(n * phi - phi_eq))**2',
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class OPLSTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='OPLSTorsionPotential',
-                 expression=('0.5 * k0 + '
-                             '0.5 * k1 * (1 + cos(phi)) + '
-                             '0.5 * k2 * (1 - cos(2*phi)) + '
-                             '0.5 * k3 * (1 + cos(3*phi)) + '
-                             '0.5 * k4 * (1 - cos(4*phi))'),
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class RyckaertBellemansTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='RyckaertBellemansTorsionPotential',
-                 expression=('c0 * cos(phi)**0 + c1 * cos(phi)**1 + '
-                             'c2 * cos(phi)**2 + c3 * cos(phi)**3 + '
-                             'c4 * cos(phi)**4 + c5 * cos(phi)**5'),
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
+            template=template,
         )
 
-class HarmonicImproperPotential(PotentialTemplate):
-    def __init__(self,
-                 name='HarmonicImproperPotetial',
-                 expression='0.5 * k * (phi - phi_eq)**2',
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
 
+class PotentialTemplates(Singleton):
```

Comment:
I'm not a fan of this name, being one character off of what it is made up on

Reply:
@mattwthompson Suggestion needed for this.

Reply:
In Monday's meeting I had suggested `PotentialTemplateLibrary` or `PotentialLibrary`

Reply:
How about `TemplatePotentialsLibrary`?